code/modules/client/preferences/preferences.dm

```diff
@@ -442,10 +442,24 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		preference.apply_to_human(character, read_character_preference(preference.type))
 
 	character.dna.real_name = character.real_name
+	if(changes_mind_name && character.mind)
+		character.mind.name = character.real_name
```

Comment:
Something feels wrong about this. It feels like it is doing more than this function should be doing when it is taking in a variable specifically to change this but doesn't always change it.
The apply prefs to function should just fully replace everything about the body for initial spawning.

Reply:
Changing your character to a specific preference should also change your mind name in general. most codes using `apply_prefs_to()` will not handle mind name properly when it's necessary.